modules/openxr/editor/openxr_action_map_editor.cpp

```diff
@@ -0,0 +1,347 @@
+	action_map->add_action_set(new_action_set);
+
+	// update our editor right away
+	_add_action_set_editor(new_action_set);
+}
+
+void OpenXRActionMapEditor::_remove_action_set(String p_name) {
+	ERR_FAIL_COND(action_map.is_null());
+	Ref<OpenXRActionSet> action_set = action_map->find_action_set(p_name);
+	ERR_FAIL_COND(action_set.is_null());
+
+	if (action_set->get_action_count() > 0) {
+		// we should remove these and add to our redo/undo step before calling _remove_action_set
+		WARN_PRINT("Action set still has associated actions before being removed!");
+	}
+
+	// now we remove it
+	action_map->remove_action_set(action_set);
+}
+
+void OpenXRActionMapEditor::_on_add_action_set() {
+	ERR_FAIL_COND(action_map.is_null());
+	String new_name = "New";
+	int count = 0;
+
+	while (action_map->find_action_set(new_name).is_valid()) {
+		new_name = "New_" + itos(count++);
+	}
+
+	_add_action_set(new_name);
+
+	// TODO set focus on this entry
+}
+
+void OpenXRActionMapEditor::_on_remove_action_set(Object *p_action_set_editor) {
+	ERR_FAIL_COND(action_map.is_null());
+
+	OpenXRActionSetEditor *action_set_editor = Object::cast_to<OpenXRActionSetEditor>(p_action_set_editor);
+	ERR_FAIL_NULL(action_set_editor);
+	ERR_FAIL_COND(action_set_editor->get_parent() != actionsets_vb);
+	Ref<OpenXRActionSet> action_set = action_set_editor->get_action_set();
+	ERR_FAIL_COND(action_set.is_null());
+
+	action_map->remove_action_set(action_set);
+	actionsets_vb->remove_child(action_set_editor);
+	action_set_editor->queue_delete();
+}
+
+void OpenXRActionMapEditor::_on_add_interaction_profile() {
+	ERR_FAIL_COND(action_map.is_null());
+
+	PackedStringArray already_selected;
+
+	for (int i = 0; i < action_map->get_interaction_profile_count(); i++) {
+		already_selected.push_back(action_map->get_interaction_profile(i)->get_interaction_profile_path());
+	}
+
+	select_interaction_profile_dialog->open(already_selected);
+}
+
+void OpenXRActionMapEditor::_on_interaction_profile_selected(const String p_path) {
+	ERR_FAIL_COND(action_map.is_null());
+
+	Ref<OpenXRInteractionProfile> new_profile;
+	new_profile.instantiate();
+	new_profile->set_interaction_profile_path(p_path);
+	action_map->add_interaction_profile(new_profile);
+
+	_add_interaction_profile_editor(new_profile);
+
+	tabs->set_current_tab(tabs->get_tab_count() - 1);
+}
+
+void OpenXRActionMapEditor::_load_action_map(const String p_path, bool p_create_new_if_missing) {
+	action_map = ResourceLoader::load(p_path, "", ResourceFormatLoader::CACHE_MODE_IGNORE);
+	if (action_map.is_null()) {
+		if (p_create_new_if_missing) {
+			action_map.instantiate();
+			action_map->create_default_action_sets();
+		} else {
+			EditorNode::get_singleton()->show_warning(TTR("Invalid file, not an OpenXR action map."));
+
+			edited_path = "";
+			header_label->set_text("");
+			return;
+		}
+	}
+
+	edited_path = p_path;
+	header_label->set_text(String(TTR("OpenXR Action map")) + ": " + p_path.get_file());
```

Comment:
You should the `:` in the string so that it can be localized. In French, we put a space before `:`, and in CJK languages they typically have a different char for it, etc. On the other hand it's good to keep the space *after* out the `TTR` (even if this should also be localized in some languages, but it's too easy to remove by mistake when translating).
```suggestion
	header_label->set_text(TTR("OpenXR Action map:") + " " + p_path.get_file());
```